lib/poetry.js

```diff
@@ -21,12 +21,27 @@ async function pyprojectTomlToRequirements(modulePath, pluginInstance) {
     generateRequirementsProgress = progress.get(
       'python-generate-requirements-toml'
     );
-    generateRequirementsProgress.update(
-      'Generating requirements.txt from "pyproject.toml"'
-    );
-    log.info('Generating requirements.txt from "pyproject.toml"');
+  }
+
+  const emitMsg = (msg) => {
+    if (generateRequirementsProgress) {
+      generateRequirementsProgress.update(msg);
+      log.info(msg);
+    } else {
+      serverless.cli.log(msg);
+    }
+  };
+
+  if (fs.existsSync('poetry.lock')) {
+    emitMsg('Generating requirements.txt from poetry.lock');
   } else {
-    serverless.cli.log('Generating requirements.txt from pyproject.toml...');
+    if (options.requirePoetryLockFile) {
+      throw new serverless.classes.Error(
+        'poetry.lock file not found - set requirePoetryLockFile to false to ' +
```

Comment:
Let's add a second argument to `Error`, which is a specific error code, maybe `MISSING_REQUIRED_POETRY_LOCK`